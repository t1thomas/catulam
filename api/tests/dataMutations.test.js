const { ApolloServer } = require('apollo-server-express');
const { createTestClient } = require('apollo-server-testing');
const driver = require('./testDb/testDbNeo4jDriver');
const gqlQueries = require('./gql-queries-mutations');
const jwt = require('jsonwebtoken');

// schema used for testing is same as production
const schema = require('../src/graphQL-schema');
// Create a test server
const server = new ApolloServer({
    context: { driver },
    schema,
});
// pass server to testing client
const { mutate } = createTestClient(server);




describe('Mutations', () => {

    it('Create 3 tickets', async () => {
        const res = await mutate({
            mutation: gqlQueries.CREATE_TICKETS,
        });
        expect(res.data).toMatchSnapshot();
    });

    it('Create 3 UserStories', async () => {
        const res = await mutate({
            mutation: gqlQueries.CREATE_USER_STORY,
        });
        expect(res.data).toMatchSnapshot();
    });

    it('Create 3 Sprints', async () => {
        const res = await mutate({
            mutation: gqlQueries.CREATE_SPRINTS,
        });
        expect(res.data).toMatchSnapshot();
    });

    it('Link 3 Tickets to UserStories', async () => {
        const res = await mutate({
            mutation: gqlQueries.ADD_USER_STORY_TICKETS,
        });
        expect(res.data).toMatchSnapshot();
    });

    it('Link 3 Tickets to Sprints', async () => {
        const res = await mutate({
            mutation: gqlQueries.ADD_SPRINT_TICKETS,
        });
        expect(res.data).toMatchSnapshot();
    });

});
