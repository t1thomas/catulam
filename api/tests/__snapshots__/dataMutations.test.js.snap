// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutations Create 3 Sprints 1`] = `
Object {
  "sprint1": Object {
    "id": "sp1test",
    "sprintNo": 1,
  },
  "sprint2": Object {
    "id": "sp2test",
    "sprintNo": 2,
  },
  "sprint3": Object {
    "id": "sp3test",
    "sprintNo": 3,
  },
}
`;

exports[`Mutations Create 3 UserStories 1`] = `
Object {
  "uStory1": Object {
    "id": "us1test",
    "storyText": "Example User Story 1",
  },
  "uStory2": Object {
    "id": "us2test",
    "storyText": "Example User Story 2",
  },
  "uStory3": Object {
    "id": "us3test",
    "storyText": "Example User Story 3",
  },
}
`;

exports[`Mutations Create 3 tickets 1`] = `
Object {
  "tick1": Object {
    "desc": "Create API for backend and frontend",
    "done": false,
    "hourEstimate": 5,
    "id": "t1test",
    "issueNumber": 1,
    "title": "Set Up API",
  },
  "tick2": Object {
    "desc": "Create Login Page for users to enter credentials",
    "done": false,
    "hourEstimate": 5,
    "id": "t2test",
    "issueNumber": 2,
    "title": "login page",
  },
  "tick3": Object {
    "desc": "Create landing page for client side SPA",
    "done": false,
    "hourEstimate": 5,
    "id": "t3test",
    "issueNumber": 3,
    "title": "Landing page",
  },
}
`;

exports[`Mutations Create 3 tickets 1`] = `
Object {
  "tick1": Object {
    "desc": "Create API for backend and frontend",
    "done": false,
    "hourEstimate": 5,
    "id": "t1test",
    "issueNumber": 1,
    "title": "Set Up API",
  },
  "tick2": Object {
    "desc": "Create Login Page for users to enter credentials",
    "done": false,
    "hourEstimate": 5,
    "id": "t2test",
    "issueNumber": 2,
    "title": "login page",
  },
  "tick3": Object {
    "desc": "Create landing page for client side SPA",
    "done": false,
    "hourEstimate": 5,
    "id": "t3test",
    "issueNumber": 3,
    "title": "Landing page",
  },
}
`;

exports[`Mutations Link 3 Tickets to Sprints 1`] = `
Object {
  "SprintTic1": Object {
    "from": Object {
      "id": "t1test",
    },
    "to": Object {
      "id": "sp1test",
    },
  },
  "SprintTic2": Object {
    "from": Object {
      "id": "t2test",
    },
    "to": Object {
      "id": "sp2test",
    },
  },
  "SprintTic3": Object {
    "from": Object {
      "id": "t3test",
    },
    "to": Object {
      "id": "sp3test",
    },
  },
}
`;

exports[`Mutations Link 3 Tickets to UserStories 1`] = `
Object {
  "UStoryTic1": Object {
    "from": Object {
      "id": "t1test",
    },
    "to": Object {
      "id": "us1test",
    },
  },
  "UStoryTic2": Object {
    "from": Object {
      "id": "t2test",
    },
    "to": Object {
      "id": "us2test",
    },
  },
  "UStoryTic3": Object {
    "from": Object {
      "id": "t3test",
    },
    "to": Object {
      "id": "us3test",
    },
  },
}
`;
